numpy/core/include/numpy/npy_3kcompat.h

```diff
@@ -62,13 +60,7 @@ static NPY_INLINE int PyInt_Check(PyObject *op) {
     PySlice_GetIndicesEx((PySliceObject *)op, nop, start, end, step, slicelength)
 #endif
 
-/* <2.7.11 and <3.4.4 have the wrong argument type for Py_EnterRecursiveCall */
-#if (PY_VERSION_HEX < 0x02070B00) || \
-    ((0x03000000 <= PY_VERSION_HEX) && (PY_VERSION_HEX < 0x03040400))
```

Comment:
This appears to be a bug(?) or at least not to match the comment.

I think it should have been

((0x03000000 >= PY_VERSION_HEX) && (PY_VERSION_HEX < 0x03040400))

Reply:
2.7.12+ was deliberately excluded from this `if` before

Reply:
Yep, you are correct.
I was reading `(0x03000000 <= PY_VERSION_HEX)` as `(PY_VERSION_HEX <= 0x03000000)`